packages/react-reconciler/src/ReactFiberCompleteWork.js

```diff
@@ -404,6 +421,20 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
     renderExpirationTime: ExpirationTime,
   ): Fiber | null {
     const newProps = workInProgress.pendingProps;
+
+    if (enableProfileModeMetrics) {
+      if (workInProgress.mode & ProfileMode) {
```

Comment:
Can you move this `resetExpirationTime` in the scheduler? That's where we "bubble up" expiration time, which is conceptually similar.

Reply:
Sure. No strong preference.